sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -415,8 +419,9 @@ public class SentryOptions {
   private boolean traceOptionsRequests = true;
 
   /** Date provider to retrieve the current date from. */
-  @ApiStatus.Internal
-  private @NotNull SentryDateProvider dateProvider = new SentryAutoDateProvider();
+  @ApiStatus.Internal private volatile @Nullable SentryDateProvider dateProvider;
```

Comment:
We could use the `LazyEvaluator`, we'd just need to add a setValue() there.
Something like 
```
  private @NotNull LazyEvaluator<SentryDateProvider> dateProvider = new LazyEvaluator<>(() -> new SentryAutoDateProvider());


  public @NotNull SentryDateProvider getDateProvider() {
    return dateProvider.getValue();
  }
  public void setDateProvider(final @NotNull SentryDateProvider dateProvider) {
    this.dateProvider.setValue(dateProvider);
  }
```
wdyt?

Reply:
yeah that sounds good, but then we'd also have to rework the LazyEvaluator to support atomicity (as in make the value volatile and do double-checked locking when initializing the value). Wdyt? Or should we introduce a new (sub)type, like AtomicLazyEvaluator that does that?

Reply:
Sounds good, should `value` in `LazyEvaluator` be `volatile`?
Should we use double checked locking, to avoid acquiring the lock for every read (then `volatile` would be required for sure)?

Reply:
yeah that would be my idea, if we implement AtomicLazyEvaluator or something

Reply:
we can change the `LazyEvaluator` directly, as the idea was to be thread safe anyway